### ISSUE TITLE
[JS] napi generator inserts `in` typemaps twice when using `numinputs=0`

### ISSUE THREAD

**Issue by mmomtchev** [swig/swig#2652]
```
ENGINE=napi make ignore_parameter.cpptest
grep -C 1 "hello" ignore_parameter_wrap.cxx
```

**Comment by wsfulton** [comment]
is this fixed by #2653?

**Comment by ojwb** [comment]
Yes (@mmomtchev I think you needs to say "Fixes #nnnn" not "Fix #nnnn" in the commit message for it to automatically get linked).

However I just noticed there's a warning from that fix:

```
Modules/javascript.cxx: In member function ‘virtual void NAPIEmitter::marshalInputArgs(Node*, ParmList*, Wrapper*, JSEmitter::MarshallingMode, bool, bool)’:
Modules/javascript.cxx:2948:5: warning: ‘tm’ may be used uninitialized [-Wmaybe-uninitialized]
 2948 |     if (tm) {
      |     ^~
Modules/javascript.cxx:2878:11: note: ‘tm’ was declared here
 2878 |   String *tm;
      |           ^~
```

Initialising `tm` to `NULL` would fix the warning, but I'm not sure that's correct - it would mean we always do `p = nextSibling(p);` after a parameter `tmap:in:numinputs` set to `0` rather than `p = Getattr(p, "tmap:in:next");` but in this case there **is** an `in` typemap - looking at other target languages it looks to me like that decision should be based on whether `tmap:in` is set or not (and it should be if `tmap:in:numinputs` is set).

**Comment by mmomtchev** [comment]
This is a real problem in all generators, it has been masked in V8/JSC because the variable is initialised.

**Comment by mmomtchev** [comment]
However, as the input typemap can't really be `NULL` - because the code will be invalid - there are no functional consequences.
